interface/modules/custom_modules/oe-module-comlink-telehealth/src/Controller/TeleHealthCalendarController.php

```diff
@@ -0,0 +1,203 @@
+        return $event;
+    }
+
+    public function addCalendarStylesheet(StyleFilterEvent $event)
+    {
+        if ($this->isCalendarPageInclude($event->getPageName())) {
+            $styles = $event->getStyles();
+            $styles[] = $this->getAssetPath() . CacheUtils::addAssetCacheParamToPath("css/telehealth.css");
+            $event->setStyles($styles);
+        }
+    }
+
+    public function renderAppointmentJavascript(AppointmentRenderEvent $event)
+    {
+        $appt = $event->getAppt();
+        // we need to grab the telehealth providers
+        $categories = $this->calendarEventCategoryRepository->getEventCategories();
+        $categoryIds = array_keys($categories);
+        $providers = $this->teleHealthProviderRepository->getEnabledProviders();
+        $providerIds = array_map(function ($provider) {
+            return intval($provider->getDbRecordId());
+        }, $providers);
+        //
+        echo $this->twig->render("comlink/appointment/add_edit_event.js.twig", ['appt' => $appt
+            , 'providers' => $providerIds, 'categories' => $categoryIds]);
+    }
+
+    public function addCalendarJavascript(ScriptFilterEvent $event)
+    {
+
+        $pageName = $event->getPageName();
+        $scriptPath = '';
+        // backwards compatible with older versions of OpenEMR
+        if (defined('OpenEMR\Events\Core\ScriptFilterEvent::CONTEXT_ARGUMENT_SCRIPT_NAME')) {
+            $scriptPath = $event->getContextArgument(ScriptFilterEvent::CONTEXT_ARGUMENT_SCRIPT_NAME) ?? '';
+        }
+        if ($this->isCalendarPageInclude($pageName)) {
+            $scripts = $event->getScripts();
+            // currently security restrictions support scripts only on the filesystem.  Translations then
+            // are being pulled from the top level comlink instead of the inner iframe.  We may need to adjust this
+            // if we run into issues.
+            $scripts[] = $this->getAssetPath() . CacheUtils::addAssetCacheParamToPath("js/telehealth-calendar.js");
+            $event->setScripts($scripts);
+        } else if ($this->isAppointmentPageInclude($pageName, $scriptPath)) {
+            $scripts = $event->getScripts();
+            $scripts[] = $this->getAssetPath() . CacheUtils::addAssetCacheParamToPath("js/telehealth-appointment.js");
+            $event->setScripts($scripts);
+        }
+    }
+
+    public function renderAppointmentsLaunchSessionButton(AppointmentRenderEvent $event)
+    {
+        $row = $event->getAppt();
+        if (empty($row['pc_eid'])) {
+            return;
+        }
+        if (empty($this->calendarEventCategoryRepository->getEventCategoryForId($row['pc_catid']))) {
+            return;
+        }
+        // don't show the launch button for a complete status
+        if ($this->apptService->isCheckOutStatus($row['pc_apptstatus'])) {
+            return;
+        }
+        echo "<script src='" . $this->getAssetPath() . "js/telehealth-calendar.js" . "'></script>";
```

Comment:
looks good place to use shiny new CacheUtils::addAssetCacheParamToPath() function

Reply:
Realized this is a redundant piece and was added prior to the injection via the Header.php script events.  I chose to remove this line instead.